libraries/plugins/apis/transaction_status_api/transaction_status_api.cpp

```diff
@@ -68,13 +68,16 @@ DEFINE_API_IMPL( transaction_status_api_impl, find_transaction )
       }
 
       // If the expiration is on or before our last irreversible block
-      auto last_irreversible_block = _db.fetch_block_by_number( last_irreversible_block_num );
-      if ( expiration <= last_irreversible_block->timestamp )
-         return {
-            .status = transaction_status::expired_irreversible
-         };
+      if ( last_irreversible_block_num > 0 )
+      {
+         auto last_irreversible_block = _db.fetch_block_by_number( last_irreversible_block_num );
+         if ( expiration <= last_irreversible_block->timestamp )
+            return {
+               .status = transaction_status::expired_irreversible
+            };
+      }
       // If our expiration is in the past
-      else if ( expiration < fc::time_point::now() )
+      if ( expiration < fc::time_point::now() )
```

Comment:
Don't check against fc::time_point::now(). Check against the head block time. What if the node gets behind by a couple blocks? The transaction could be contained in a block that has a head time in between the head block time and now.